tlatools/org.lamport.tlatools/test/tlc2/util/BufferedRandomAccessFileFuzzTest.java

```diff
@@ -0,0 +1,553 @@
+        Object outputLock = new Object();
+
+        List<Thread> threads = new ArrayList<>(nthreads);
+        for (int i = 0; i < nthreads; ++i) {
+            final int threadID = i;
+            Thread t = new Thread(() -> {
+                Random rng = new Random(threadID);
+                while (running.get()) {
+                    int runID = nextOp.getAndIncrement();
+                    if (runID > OPS) {
+                        return;
+                    }
+                    synchronized (outputLock) {
+                        System.out.println("---- starting run " + runID);
+                    }
+                    try {
+                        RunResult result = checkOps(generateRandomOps(rng, 50));
+                        if (!result.ok()) {
+                            badResult.set(result);
+                            running.set(false);
+                        }
+                    } catch (Throwable e) {
+                        error.set(e);
+                        running.set(false);
+                        return;
+                    }
+                }
+            });
+            t.setName(getClass().getSimpleName() + "-worker-" + threadID);
+            t.setDaemon(true);
+            t.start();
+            threads.add(t);
+        }
+
+        for (Thread t : threads) {
+            try {
+                t.join();
+            } catch (InterruptedException ignored) {
+            }
+        }
+
+        RunResult result = badResult.get();
+        if (result != null && !result.ok()) {
+            System.out.println("Violation found on op # " + result.successes + "; minimizing...");
+            java.util.List<FileOperation<?>> minimalOps = minimize(
+                    result.ops.subList(0, result.successes + 1),
+                    (subset) -> legal(subset) && !checkOps(subset).ok());
+            System.out.println("Minimal trace:");
+            for (FileOperation<?> op : minimalOps) {
+                System.out.println(" --> " + op);
+            }
+            RunResult minResult = checkOps(minimalOps);
+            throw new RuntimeException("got " + minResult.actual + " but expected " + minResult.expected);
+        }
+
+        Throwable thrown = error.get();
+        if (thrown != null) {
+            throw new RuntimeException(thrown);
+        }
+    }
+
+    private java.util.List<FileOperation<?>> generateRandomOps(Random rng, int nOps) {
+        java.util.List<FileOperation<?>> ops = new ArrayList<>(nOps);
+        RandomOperationGenerator gen = new RandomOperationGenerator(rng);
+        for (int i = 0; i < nOps; ++i) {
+            ops.add(gen.nextRandomOperation());
+        }
+        if (!legal(ops)) {
+            throw new IllegalStateException("generated illegal ops: " + ops);
+        }
+        return ops;
+    }
+
+    private void generateRandomOpsAndCheck(Random rng, int nOps) throws IOException {
```

Comment:
This function appears to be redundant.  Even if it had some value in debugging (taking a guess at its purpose), I still think it makes more sense to remove it given that it has otherwise duplicated code.

Reply:
Good catch; this is leftover from some intermediate version and should be removed.